k8s/upgrades/0.8.1-0.8.2/cstor_volume_upgrade.sh

```diff
@@ -0,0 +1,175 @@
+#!/bin/bash
+
+################################################################
+# STEP: Get Persistent Volume (PV) name as argument            #
+#                                                              #
+# NOTES: Obtain the pv to upgrade via "kubectl get pv"         #
+################################################################
+volume_upgrade_version="v0.8.x-ci"
+volume_current_version="0.8.1"
+
+function usage() {
+    echo
+    echo "Usage:"
+    echo
+    echo "$0 <pv-name> <openebs-namespace>"
+    echo
+    echo "  <pv-name> Get the PV name using: kubectl get pv"
+    echo "  <openebs-namespace> Get the namespace where openebs"
+    echo "  pods are installed"
+    exit 1
+}
+
+##Checking the version of OpenEBS ####
+function verify_volume_version() {
+    local resource=$1
+    local name_res=$2
+    local openebs_version=$(kubectl get $resource $name_res -n $ns \
+                 -o jsonpath="{.metadata.labels.openebs\.io/version}")
+
+    if [[ $openebs_version != $volume_current_version ]] && [[ $openebs_version != $volume_upgrade_version ]]; then
+        echo "Expected version of $name_res in $resource is $volume_current_version but got $openebs_version";exit 1;
+    fi
+    echo $openebs_version
+}
+
+if [ "$#" -ne 2 ]; then
+    usage
+fi
+
+pv=$1
+ns=$2
+
+
+## Get storageclass and PVC related details to patch target service
+sc_ns=`kubectl get pv $pv -o jsonpath="{.spec.claimRef.namespace}"`
+sc_name=`kubectl get pv $pv -o jsonpath="{.spec.storageClassName}"`
+sc_res_ver=`kubectl get sc $sc_name -n $sc_ns -o jsonpath="{.metadata.resourceVersion}"`
+pvc_name=`kubectl get pv $pv -o jsonpath="{.spec.claimRef.name}"`
+pvc_namespace=`kubectl get pv $pv -o jsonpath="{.spec.claimRef.namespace}"`
+
+# Check if CASType is cstor for PV
+cas_type=`kubectl get pv $pv -o jsonpath="{.metadata.labels.openebs\.io/cas-type}"`
+if [ $cas_type != "cstor" ]; then
+    echo "Cstor volume not found";exit 1;
+fi
+
+### 1. Get the cstorvolume name related to the given PV ###
+### get the cloned volume cstorvolume name if exists
+echo "Upgrading Cstor Volume resource $volume_upgrade_version"
+cv_name=$(kubectl get cvr -n openebs\
+           -l openebs.io/persistent-volume=$pv\
+           -o 'jsonpath={.items[?(@.metadata.labels.openebs\.io/cloned=="true")].metadata.annotations.openebs\.io/source-volume}' | awk '{print $1}')
+
+version=$(verify_volume_version "cstorvolume" $pv)
```

Comment:
get cv from pv using command like `c_vol=$(kubectl get cstorvolumes -l openebs.io/persistent-volume=$pv -n $ns -o jsonpath="{.items[*].metadata.name}")`
don't assume pv is cv